app/models/exchange_rate_currency_rate.rb

```diff
@@ -3,6 +3,7 @@
 class ExchangeRateCurrencyRate < Sequel::Model
   SCHEDULED_RATE_TYPE = 'scheduled'.freeze
   SPOT_RATE_TYPE = 'spot'.freeze
+  AVERAGE_RATE_TYPE = 'average'.freeze
```

Comment:
Not sure if this can be done but its a same we cant group these constants into `RATE_TYPE_CONSTANTS` in case more are added or we can pass that group of constant around.. probably a simple way but brain not working right now or its over engineering.

Reply:
We could move to using an enum but I'd argue we can do that in a separate PR.

We have a couple of options, there:

1. Build a simple enum plugin which includes validations of whether the rate type value matches one of the available options and adds convenience methods `average?`
2. Add an enum type to the database and use that in the rate type implementation. This would be painful to alter/extend in future but would guarantee that we're meeting the contract we've laid out